scripts/test-upgrade.sh

```diff
@@ -18,34 +18,65 @@
 
 set -eux -o pipefail
 
-BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
-OA_DIR="$BASE_DIR/openstack-ansible"
-RPCD_DIR="$BASE_DIR/rpcd"
+export BASE_DIR=$( cd "$( dirname ${0} )" && cd ../ && pwd )
+export OA_DIR="$BASE_DIR/openstack-ansible"
+export RPCD_DIR="$BASE_DIR/rpcd"
+
+source ${BASE_DIR}/scripts/functions.sh
+
+# Create a backup of /etc/openstack_deploy, if one doesn't already exist
+# NOTE: This gets done by openstack-ansible's scripts/run-upgrade.sh script,
+#       but if we wait until then it'll be too late.
+if [[ ! -d "/etc/openstack_deploy.MITAKA" ]]; then
```

Comment:
Good thinking. This is actually [already done in the pre-upgrade playbooks](https://github.com/rcbops/rpc-openstack/blob/master/rpcd/playbooks/roles/rpc_pre_upgrade/tasks/backup.yml), so it can be removed.